File: app/src/main/java/com/wilinz/yuetingmusic/ui/main/home/TopListPlayListAdapter.java
package com.wilinz.yuetingmusic.ui.main.home;

import android.annotation.SuppressLint;
import android.net.Uri;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.wilinz.yuetingmusic.data.model.TopList;
import com.wilinz.yuetingmusic.data.model.TopListSong;
import com.wilinz.yuetingmusic.databinding.ItemFirstThreeListBinding;
import com.wilinz.yuetingmusic.databinding.ItemTopListBinding;

import java.util.List;

import kotlin.collections.CollectionsKt;


public class TopListPlayListAdapter extends RecyclerView.Adapter<TopListPlayListAdapter.MyViewHolder> {

    private List<TopListSong.PlaylistBean.TracksBean> songs;
    private OnItemClickListener listener;

    interface OnItemClickListener {
        void onItemClick(List<TopListSong.PlaylistBean.TracksBean> songs, int index, TopListSong.PlaylistBean.TracksBean song);
    }

    public TopListPlayListAdapter(@NonNull List<TopListSong.PlaylistBean.TracksBean> musics) {
        this.songs = musics;
    }

    public void set(List<TopListSong.PlaylistBean.TracksBean> songs) {
        this.songs = songs;
        notifyDataSetChanged();
    }

    public void setOnItemClickListener(OnItemClickListener l) {
        listener = l;
    }

    static class MyViewHolder extends RecyclerView.ViewHolder {

        ItemFirstThreeListBinding binding;

        public MyViewHolder(@NonNull ItemFirstThreeListBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        ItemFirstThreeListBinding binding = ItemFirstThreeListBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false);
        MyViewHolder holder = new MyViewHolder(binding);

        binding.getRoot().setOnClickListener(v -> {
            if (listener != null) {
                int index = holder.getAbsoluteAdapterPosition();
                listener.onItemClick(songs, index, songs.get(index));
            }
        });
        return holder;
    }

    @SuppressLint("SetTextI18n")
    @Override
    public void onBindViewHolder(@NonNull TopListPlayListAdapter.MyViewHolder holder, int position) {
        ItemFirstThreeListBinding binding = holder.binding;
        TopListSong.PlaylistBean.TracksBean track = songs.get(position);
        binding.serialNumber.setText((position + 1) + "");
        binding.songName.setText(track.name);

        Glide.with(binding.songAvatar)
                .load(track.al.picUrl)
                .into(binding.songAvatar);

        TopListSong.PlaylistBean.TracksBean.ArBean artist = CollectionsKt.firstOrNull(track.ar);
        if (artist != null) binding.songArtist.setText(artist.name);
    }

    @Override
    public int getItemCount() {
        int size = songs.size();
        return size;
    }


}

File: app/src/main/java/com/wilinz/yuetingmusic/ui/main/home/TopListPlayListAdapter.kt
package com.wilinz.yuetingmusic.ui.main.home

import android.annotation.SuppressLint
import android.app.Application
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.wilinz.yuetingmusic.data.model.TopListSong.PlaylistBean.TracksBean
import com.wilinz.yuetingmusic.databinding.ItemMusicBinding
import com.wilinz.yuetingmusic.util.ScreenUtil

class TopListPlayListAdapter(private var songs: List<TracksBean>) :
    RecyclerView.Adapter<TopListPlayListAdapter.MyViewHolder>() {
    private var listener: ((songs: List<TracksBean>, index: Int, song: TracksBean) -> Unit)? = null

    interface OnItemClickListener {
        fun onItemClick(songs: List<TracksBean>?, index: Int, song: TracksBean?)
    }

    fun set(songs: List<TracksBean>) {
        this.songs = songs
        notifyDataSetChanged()
    }

    fun setOnItemClickListener(l: (songs: List<TracksBean>, index: Int, song: TracksBean) -> Unit) {
        listener = l
    }

    class MyViewHolder(var binding: ItemMusicBinding) : RecyclerView.ViewHolder(
        binding.root
    )

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val binding = ItemMusicBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        val holder = MyViewHolder(binding)
        binding.root.setOnClickListener { v: View? ->
            if (listener != null) {
                val index = holder.absoluteAdapterPosition
                listener?.invoke(songs, index, songs[index])
            }
        }
        return holder
    }

    @SuppressLint("SetTextI18n")
    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val binding = holder.binding
        val track = songs[position]
        binding.serialNumber.text = (position + 1).toString() + ""
        binding.name.text = track.name
        val size = ScreenUtil.dpToPx(binding.songAvatar.context, 48)
        Glide.with(binding.songAvatar)
            .load(track.al!!.picUrl + "?param=${size}y${size}")
            .into(binding.songAvatar)
        val artist = track.ar!!.firstOrNull()
        if (artist != null) binding.secondName.text = artist.name
    }

    override fun getItemCount(): Int {
        return songs.size
    }
}